### PR TITLE
clean up protocol interface 1

### PR DESCRIPTION
**Pull request by CoderZhi**
delete aborter

### CODE REVIEW

File: state/factory/workingset.go [hunk 1]
```diff
@@ -183,17 +183,16 @@ func (ws *workingSet) Revert(snapshot int) error {
...
-	err := ws.commitFunc(ws.height)
-	if !ws.Dirty() {
+	if err := ws.commitFunc(ws.height); err != nil {
 		return err
 	}
-	if err == nil {
-		protocolCommit(ctx, ws)
-	} else {
-		protocolAbort(ctx, ws)
+	if !ws.dock.Dirty() {
```
Comment (yutongp) [inline state/factory/workingset.go:189]: i am confused here.
i thought it should be if dirty() commit

File: state/factory/workingset.go [hunk 2]
```diff
@@ -183,17 +183,16 @@ func (ws *workingSet) Revert(snapshot int) error {
 
 // Commit persists all changes in RunActions() into the DB
 func (ws *workingSet) Commit(ctx context.Context) error {
-	err := ws.commitFunc(ws.height)
-	if !ws.Dirty() {
+	if err := ws.commitFunc(ws.height); err != nil {
```
Comment (yutongp) [inline state/factory/workingset.go:186]: also seems in previous logic, ws.Reset is always called. but now it doesn't call ws.Reset on error

File: state/factory/workingset.go [hunk 3]
```diff
@@ -183,17 +183,16 @@ func (ws *workingSet) Revert(snapshot int) error {
...
 		return err
 	}
-	if err == nil {
-		protocolCommit(ctx, ws)
-	} else {
-		protocolAbort(ctx, ws)
+	if !ws.dock.Dirty() {
+		if err := protocolCommit(ctx, ws); err != nil {
```
Comment (yutongp) [inline state/factory/workingset.go:190]: also seems in previous logic, ws.Reset is called. but now it doesn't call ws.Reset on error

File: state/factory/workingset.go (hunk 1 above)
Comment (CoderZhi) [reply state/factory/workingset.go:189]: this is a bug, fixing it.

File: state/factory/workingset.go (hunk 3 above)
Comment (CoderZhi) [reply state/factory/workingset.go:190]: We actually do not need to call "Reset". The data stored in ws.dock should be similar to ws itself, which doesn't need to be explicitly cleaned up.

**Pull request closed without merging**